lasagne/layers/shape.py

```diff
@@ -119,6 +132,14 @@ def get_output_shape_for(self, input_shape, **kwargs):
                         # it is unknown.
                         masked_input_shape[o[0]] = 1
                         masked_output_shape[dim] = 1
+
+        # Secondly, replace all symbolic shapes with `None`, as we cannot
+        # infer their size here.
+        for dim, o in enumerate(output_shape):
+            if isinstance(o, T.TensorVariable):
+                output_shape[dim] = None
+                masked_output_shape[dim] = None
+
```

Comment:
Nitpicking: Please either remove the blank lines or insert them between all the sections. I found the comments to stand out enough to delineate the sections, though.